File: extension/src/lib/voices.ts
// Mirror of the backend voice pool (backend/app/services/voices.py).
// If you add/remove voices here, update the backend too.

export interface Voice {
  id: string;
  label: string;
}

// Sentinel value stored in voiceOverrides when the user picks "Mute" for a
// character. The content script skips rendering cues whose voice resolves to
// this value.
export const MUTE_VOICE_ID = "__anime_dub_mute__";

export const ELEVENLABS_VOICES: Voice[] = [
  { id: "21m00Tcm4TlvDq8ikWAM", label: "Rachel · calm female" },
  { id: "AZnzlk1XvdvUeBnXmlld", label: "Domi · strong female" },
  { id: "EXAVITQu4vr4xnSDxMaL", label: "Bella · soft female" },
  { id: "MF3mGyEYCl7XYWbV9V6O", label: "Elli · emotional female" },
  { id: "ErXwobaYiN019PkySvjV", label: "Antoni · well-rounded male" },
  { id: "TxGEqnHWrfWFTfGW9XjX", label: "Josh · deep male" },
  { id: "VR6AewLTigWG4xSOukaG", label: "Arnold · crisp male" },
  { id: "pNInz6obpgDQGcFmaJgB", label: "Adam · deep male" },
  { id: "yoZ06aMxZJJ28mfd3POQ", label: "Sam · raspy male" },
];

export const SAY_VOICES: Voice[] = [
  { id: "Samantha", label: "Samantha · female" },
  { id: "Alex", label: "Alex · male" },
  { id: "Daniel", label: "Daniel · UK male" },
  { id: "Karen", label: "Karen · AU female" },
  { id: "Moira", label: "Moira · IE female" },
  { id: "Tessa", label: "Tessa · ZA female" },
  { id: "Veena", label: "Veena · IN female" },
  { id: "Fred", label: "Fred · male" },
  { id: "Albert", label: "Albert · male" },
];

File: backend/app/services/voices.py
"""Speaker -> voice mapping. Deterministic hash -> pool index, so the same
character always gets the same voice within a session."""

import hashlib

ELEVENLABS_VOICE_POOL: list[str] = [
    "21m00Tcm4TlvDq8ikWAM",  # Rachel  - calm female
    "AZnzlk1XvdvUeBnXmlld",  # Domi    - strong female
    "EXAVITQu4vr4xnSDxMaL",  # Bella   - soft female
    "MF3mGyEYCl7XYWbV9V6O",  # Elli    - emotional female
    "ErXwobaYiN019PkySvjV",  # Antoni  - well-rounded male
    "TxGEqnHWrfWFTfGW9XjX",  # Josh    - deep male
    "VR6AewLTigWG4xSOukaG",  # Arnold  - crisp male
    "pNInz6obpgDQGcFmaJgB",  # Adam    - deep male
    "yoZ06aMxZJJ28mfd3POQ",  # Sam     - raspy male
]

# macOS `say -v ?` voices that ship with every recent macOS install
SAY_VOICE_POOL: list[str] = [
    "Samantha",
    "Alex",
    "Daniel",
    "Karen",
    "Moira",
    "Tessa",
    "Veena",
    "Fred",
    "Albert",
]


def _hash_index(speaker: str, pool_size: int) -> int:
    h = hashlib.sha1(speaker.encode("utf-8")).digest()
    return int.from_bytes(h[:4], "big") % pool_size


def pick_elevenlabs_voice(speaker: str | None, default: str) -> str:
    if not speaker:
        return default
    return ELEVENLABS_VOICE_POOL[_hash_index(speaker, len(ELEVENLABS_VOICE_POOL))]


def pick_say_voice(speaker: str | None) -> str | None:
    if not speaker:
        return None
    return SAY_VOICE_POOL[_hash_index(speaker, len(SAY_VOICE_POOL))]
